dot/telemetry/telemetry.go

```diff
@@ -124,31 +97,94 @@ func (h *Handler) startListening() {
 		go func() {
 			for _, conn := range h.connections {
 				conn.Lock()
-				err := conn.wsconn.WriteMessage(websocket.TextMessage, msgToBytes(msg))
+				defer conn.Unlock()
+				msgBytes, err := h.msgToJSON(msg)
+				if err != nil || len(msgBytes) == 0 {
+					h.log.Debug("issue decoding telemetry message", "error", err)
+					return
+				}
```

Comment:
move this out of the for loop

Reply:
good idea, no need to keep rebuilding the message.